### DISCUSSION TITLE
NVidia Kaolin, Anaconda Python setup, and CUDA version: Beginner’s hurdles

### DISCUSSION THREAD

**Discussion by tahasid2021** [NVIDIAGameWorks/kaolin#633] [Q&A] [answered]
I saw, in other post, this particular problem is already resolved in newer versions of Kaolin. But unfortunately I am stuck with that. I am assuming this can be a mistake from my end. 
Please help me to understand where I am doing wrong. My anaconda setup is not recognizing torch version and CUDA version, even though they are installed correctly in  C:\ProgramData\Anaconda3\envs\kaolin\Lib\site-packages\.
I am trying to install torch 1.12.1 with CUDA 11.3 as instructed in the installation documentation of Kaolin. 

I am trying to run kaolin with anaconda and with Jupiter notebook. I had installed CUDA toolkit 11.8 before installing anaconda and it worked fine, that means gpu is working perfect.

Basic anaconda installation as below:
MY CONDA COMMANDS STEP BY STEP AS BELOW, in case I am missing something.:
ENVIRONMENTAL PATH VERIABLES FOR ANACONDA ADDED AS BELOW:
C:\ProgramData\anaconda3\Library\bin\conda.bat
C:\ProgramData\anaconda3\Scripts\conda.exe
C:\ProgramData\anaconda3\condabin\conda.bat

ENVIRONMENTAL PATH VARIABLES FOR GIT IS ADDED AS BELOW:
C:\Program Files\Git\bin
C:\Program Files\Git\cmd

CONDA FORGE CHANNEL SETUP:
conda config –add channels conda-forge
conda config –set channel_priority strict

CONDA KAOLIN ENVIRONMENT SETUP:
conda create –name kaolin python=3.9
conda activate kaolin

GO TO THE FOLDER
C:\Users\**\Documents\Projects

CLONE GIT REPOSITORY
git clone –recursive https://github.com/NVIDIAGameWorks/kaolin.git

#then go to Kaolin directory:

cd kaolin

git checkout v0.12.0
#YOU ARE IN ‘DETACHED HEAD’ STATE. 

#Now start using master branch

git checkout master

ADD ENVIRONMENT VARIABLES, THIS STEP IS VERY IMPORTANT. I personally think I did something wrong here. Please correct me.

set USE_CYTHON=1 
set IGNORE_TORCH_VER=1
set KAOLIN_INSTALL_EXPERIMENTAL=1 
set TORCH_CUDA_ARCH_LIST=11.3

# now install torch, torchvision, torchaudio from CUDA 11.3 environment.

pip install torch==1.12.1 torchvision==0.13.1 torchaudio==0.12.1 --extra-index-url https://download.pytorch.org/whl/cu113

Defaulting to user installation because normal site-packages is not writeable
Looking in indexes: https://pypi.org/simple, https://download.pytorch.org/whl/cu113
Requirement already satisfied: torch==1.12.1 in c:\programdata\anaconda3\envs\kaolin\lib\site-packages (1.12.1+cu113)
Requirement already satisfied: torch==1.12.1 in c:\programdata\anaconda3\envs\kaolin\lib\site-packages (1.12.1+cu113)
Requirement already satisfied: torchvision==0.13.1 in c:\programdata\anaconda3\envs\kaolin\lib\site-packages (0.13.1+cu113)
Requirement already satisfied: torchaudio==0.12.1 in c:\programdata\anaconda3\envs\kaolin\lib\site-packages (0.12.1+cu113)
Requirement already satisfied: typing-extensions in c:\programdata\anaconda3\envs\kaolin\lib\site-packages (from torch==1.12.1) (4.4.0)
Requirement already satisfied: requests in c:\programdata\anaconda3\envs\kaolin\lib\site-packages (from torchvision==0.13.1) (2.28.1)
Requirement already satisfied: numpy in c:\programdata\anaconda3\envs\kaolin\lib\site-packages (from torchvision==0.13.1) (1.23.3)
Requirement already satisfied: pillow!=8.3.*,>=5.3.0 in c:\programdata\anaconda3\envs\kaolin\lib\site-packages (from torchvision==0.13.1) (9.2.0)
Requirement already satisfied: certifi>=2017.4.17 in c:\programdata\anaconda3\envs\kaolin\lib\site-packages (from requests->torchvision==0.13.1) (2022.9.24)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in c:\programdata\anaconda3\envs\kaolin\lib\site-packages (from requests->torchvision==0.13.1) (1.26.12)
Requirement already satisfied: charset-normalizer<3,>=2 in c:\programdata\anaconda3\envs\kaolin\lib\site-packages (from requests->torchvision==0.13.1) (2.1.1)
Requirement already satisfied: idna<4,>=2.5 in c:\programdata\anaconda3\envs\kaolin\lib\site-packages (from requests->torchvision==0.13.1) (3.4)

errors as below. Torch is not recognized. nvcc CUDA version is showing 11.8, which is installed in my computer, not in anaconda environment.

(kaolin) C:\Users\**\Documents\Projects\kaolin>torch –version
‘torch’ is not recognized as an internal or external command,
operable program or batch file.

(kaolin) C:\Users\TH\Documents\Projects\kaolin>nvcc –version
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2022 NVIDIA Corporation
Built on Wed_Sep_21_10:41:10_Pacific_Daylight_Time_2022
Cuda compilation tools, release 11.8, V11.8.89
Build cuda_11.8.r11.8/compiler.31833905_0

I also tried to install via conda 
I also installed via conda
conda install -c pytorch pytorch=1.12.1 torchvision cudatoolkit=11.3
but getting same error.

Please let me know what I am doing wrong. I am from c# dot net background, so I can miss something very important.

**Comment by Caenorst** [comment, marked as answer]
Hi @tahasid2021 , thank you for your interest in Kaolin.

Here is one thing I can notice, `TORCH_CUDA_ARCH_LIST=11.3` for sure is wrong as `11.3` is NOT an architecture. Please unset this variable as this is typically a value you would set in exotic edge cases of cross-compilation.

> (kaolin) C:\Users**\Documents\Projects\kaolin>torch –version

torch is not a command-line, this is a python module, if you want to know the version run the following:
```
python -c "import torch; torch.__version__"
```

**Reply by tahasid2021** [reply]
Thanks for the reply. What should I write instead of, TORCH_CUDA_ARCH_LIST=11.3. Do I need to do a fresh install?

**Reply by tahasid2021** [reply]
Sorry for novice question 
:)

**Reply by Caenorst** [reply]
Just unset the variable before installing Kaolin, there is nothing to write instead.

So far the only error you are sharing is:
```
(kaolin) C:\Users**\Documents\Projects\kaolin>torch –version
‘torch’ is not recognized as an internal or external command,
operable program or batch file.
```
Is there anything else wrong?

**Reply by tahasid2021** [reply]
Sorry for the delay, Thanks for letting me know. I was out of my work station. I will write you back few hours later.

**Comment by tahasid2021** [comment]
<python -c "import torch; torch.__version__">
did not give any output.  Not sure why.

So I used 
<conda list -f pytorch>
It gave this output below.

<# packages in environment at C:\ProgramData\Anaconda3\envs\kaolin:>

 <#Name                    Version                   Build  Channel>
<#pytorch                   1.12.1          py3.9_cuda11.3_cudnn8_0    pytorch>

I am going through another deadline, so delayed to test it.  Thanks!
